design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +235,134 @@ permissionScope: Cluster
 
 ### Stack Install
 
-Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to allow differing access to request their installation.
-A simple way to approach this would be to create different installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to
+ allow differing access to request their installation. A simple way to approach this would be to create different
+ installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+
 Currently, we use a single `StackRequest` type, but going forward we should differentiate by stack type:
 
-* `ClusterStackInstall`: requests the installation of a cluster stack
-* `StackInstall`: requests the installation of a namespace stack
+* `ClusterStackInstall`: requests the installation of a cluster stack who's resources are all cluster scoped
```

Comment:
> who's resource are all clusters scoped

So this translates into the Stacks Manager only needing to create / update `ClusterRole` definitions for the environment admin, editor, viewer roles, as part of processing a `ClusterStackInstall`?

Reply:
I think even for a namespaced StackInstall, environment admins, editors, and viewers should have access to interact with the namespaced resources where ever they are found.  Clusterroles for the stack installed types should aggregate to the roles used by these environment personas.

Reply:
I think it would be good to clear this up. When I hear separation of concerns, I hear two different groups of people without overlap, thus separation. I think if cluster-admin wants to make someone a crossplane-admin they would have access to admin everything, but an environment-admin would only be able to admin the environment or infrastructure related resources and config.

We haven't specified if cluster-admin will get aggregated roles from the namespace stack installs, but since cluster-admin will be cluster scoped, this would potentially allow a namespaced stack to grant the cluster-admin to write the newly installed namespace scoped resource into any namespace, which would be unideal.

Reply:
> this would potentially allow a namespaced stack to grant the cluster-admin to write the newly installed namespace scoped resource into any namespace, which would be unideal

Can't the cluster-admin do that anyway? If they want to create a resource in a namespace where no controller will act on them - can't they do that, if they want?